Security/OAuthUtils.php

```diff
@@ -51,14 +52,15 @@ class OAuthUtils
     protected $securityContext;
 
     /**
-     * @param HttpUtils                $httpUtils
-     * @param SecurityContextInterface $securityContext
-     * @param boolean                  $connect
+     * @param HttpUtils                     $httpUtils
+     * @param SecurityContextInterface      $securityContext
+     * @param boolean                       $connect
+     * @param AuthorizationCheckerInterface $authChecker
      */
-    public function __construct(HttpUtils $httpUtils, SecurityContextInterface $securityContext, $connect)
+    public function __construct(HttpUtils $httpUtils, SecurityContextInterface $securityContext = null, $connect, AuthorizationCheckerInterface $authChecker = null)
```

Comment:
Don't add extra argument. Only `isGranted` method is needed here.

Just rename `securityContext` to `authorizationChecker` without class type on signature and check argument manually.

You can have a sample here: https://github.com/sonata-project/SonataAdminBundle/pull/2963/files#diff-63617c580370c072c42af8368dfc9e21R40